src/lib.rs

```diff
@@ -58,11 +58,17 @@ pub async fn run_trin(
         setup_temp_dir();
     }
 
-    let storage_config =
-        PortalStorageConfig::new(trin_config.kb.into(), discovery.local_enr().node_id());
+    let storage_config = PortalStorageConfig::new(
+        trin_config.kb.into(),
+        discovery.local_enr().node_id(),
+        trin_config.enable_metrics_with_url.is_some(),
```

Comment:
At this point, is it worth just passing in `trin_config` directly?

Reply:
I'd say yes if `PortalStorageConfig` was using 1.) all of the fields of `TrinConfig` and 2.) only the fields of `TrinConfig`, otherwise I think that would make this `new` function much more opaque as far as what's being passed, and much less flexible for future changes.